src/docfx/build/xref/InternalXrefMapBuilder.cs

```diff
@@ -132,22 +132,28 @@ private InternalXrefSpec[] AggregateXrefSpecs(string uid, InternalXrefSpec[] spe
                 specsWithSameUid = specsWithSameUid.Where(spec => spec.DeclaringFile.Origin != FileOrigin.Fallback).ToArray();
             }
 
-            // multiple uid conflicts without moniker range definition
+            // multiple uid conflicts
             // log an warning and take the first one order by the declaring file
-            var duplicatedSpecs = specsWithSameUid.Where(item => item.Monikers.Count == 0).ToArray();
-            if (duplicatedSpecs.Length > 1)
+            var duplicatedSpecs = specsWithSameUid.GroupBy(spec => spec.Monikers);
+            var duplicateMonikerList = new HashSet<MonikerList>();
+            foreach (var monikerGroup in duplicatedSpecs)
             {
-                var duplicatedSources = (from spec in duplicatedSpecs where spec.Uid.Source != null select spec.Uid.Source).ToArray();
-                foreach (var spec in duplicatedSpecs)
+                var specsWithSameMonikerList = monikerGroup.ToList();
+                var duplicateSource = (from spec in specsWithSameMonikerList where spec.Uid.Value != null select spec.Uid.Source).ToArray();
+                if (specsWithSameMonikerList.Count > 1)
```

Comment:
move this line before line 142 to not create `duplicateSource` in case `Count <= 1`.

Reply:
fixed